days/src/day1/Day1.kt

```diff
@@ -0,0 +1,34 @@
+package day1
+
+import Day
+import java.io.File
+import kotlin.test.assertEquals
+
+class Day1(private val expenseReport: List<Int>): Day {
+    override fun puzzle1() = expenseReport.asSequence().associateBy { 2020 - it }.let { adjusted ->
+        val result = expenseReport.filter { adjusted.contains(it) }
```

Comment:
The solution looks good, but I did have two Kotlin specific questions:

1. Does the `associatedBy` function on the sequence (and `let` that follows it) execute in order? Meaning does the entire line (including the `let`) execute for each item in the list or does the `associatedBy` complete before any of the `let` portions execute?
2. Which part of the `filter` function causes it to return a `first` and `last`?

Reply:
I think the `let` is final so the `asSequence` doesn't really do anything here. 
the filter will just return the only solution. Assuming there is only one (I believe that was in the problem). Otherwise this would not work. The `first()` and `last()` below just take those two values